### PR TITLE
Add caching for sanitized labels in the Prometheus MetricAdapter.

### LINKED ISSUE #1341: Clean up the label processing logic in the prometheus MetricAdapter
There's a TODO in the code for caching sanitized labels, for instance.

### PR DESCRIPTION
**Pull request by breedx-splk**
This resolves #1341.

### CODE REVIEW

**Review by jkwatson — APPROVED** [review]
Thanks!

File: exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/LabelNameSanitizerTest.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.function.Function;
+import org.junit.jupiter.api.Test;
+
+class LabelNameSanitizerTest {
+
+  @Test
+  void testSanitizerCaching() {
+    AtomicInteger count = new AtomicInteger();
+    Function<String, String> delegate = labelName -> labelName + count.incrementAndGet();
+    LabelNameSanitizer testClass = new LabelNameSanitizer(delegate);
```
Comment (anuraaga) [inline exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/LabelNameSanitizerTest.java:20]:
```suggestion
    LabelNameSanitizer sanitizer = new LabelNameSanitizer(delegate);
```

makes the test easier to read

File: exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/LabelNameSanitizer.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ */
+
+package io.opentelemetry.exporter.prometheus;
+
+import io.prometheus.client.Collector;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/** Used to convert a label keys to a label names. Sanitizes the label keys. Not thread safe. */
```
[the remark this answers is not in the document]
Comment (breedx-splk) [reply exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/LabelNameSanitizer.java:13]: I don't know the answer, but I'm cool with using `ConcurrentHashMap` here as a safety precaution.  Thanks for the suggestion.

File: exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/LabelNameSanitizerTest.java (hunk 1 above)
Comment (breedx-splk) [reply exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/LabelNameSanitizerTest.java:20]: Sure thing.  We can talk about the pros/cons of each approach sometime, but I'm happy to change it here.

**Comment by anuraaga** [comment]
LGTM, thanks1

**Pull request merged**